diff --git a/.htaccess b/.htaccess
index 6f9123d14..67f9bd2f4 100644
--- a/.htaccess
+++ b/.htaccess
@@ -183,3 +183,14 @@ AddEncoding gzip svgz
   # Disable Proxy header, since it's an attack vector.
   RequestHeader unset Proxy
 </IfModule>
+
+<ifmodule mod_setenvif.c>
+  SetEnvIf AH_CLIENT_IP ^45\.79\.182\.167$ DENY=1
+  SetEnvIf AH_CLIENT_IP ^45\.79\.149\.122$ DENY=1
+  SetEnvIf AH_CLIENT_IP ^45\.33\.86\.146$ DENY=1
+  SetEnvIf AH_CLIENT_IP ^96\.126\.107\.129$ DENY=1
+  SetEnvIf AH_CLIENT_IP ^96\.126\.107\.119$ DENY=1
+  Order allow,deny
+  Allow From All
+  Deny from env=DENY
+</ifmodule>
