diff --git a/src/TBMegaMenuBuilder.php b/src/TBMegaMenuBuilder.php
index 11cce22..ea4d773 100644
--- a/src/TBMegaMenuBuilder.php
+++ b/src/TBMegaMenuBuilder.php
@@ -534,7 +534,9 @@ class TBMegaMenuBuilder {
       }
     }
     for ($j = count($col_content); $j > $i; $j--) {
-      $item_config['rows_content'][$row][$col]['col_content'][$j] = $item_config['rows_content'][$row][$col]['col_content'][$j - 1];
+      if(isset($item_config['rows_content'][$row][$col]['col_content'][$j - 1])) {
+        $item_config['rows_content'][$row][$col]['col_content'][$j] = $item_config['rows_content'][$row][$col]['col_content'][$j - 1];
+      }
     }
     $item_config['rows_content'][$row][$col]['col_content'][$i] = [
       'plugin_id' => $item->link->getPluginId(),
